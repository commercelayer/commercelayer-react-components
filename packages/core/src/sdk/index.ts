import sdk, { type CommerceLayerClient } from "@commercelayer/sdk"
import { jwtDecode } from "jwt-decode"
import type { Jwt, RequestConfig } from "src/types/index.js"

export function getSdk({ accessToken }: RequestConfig): CommerceLayerClient {
  const organization = jwtDecode<Jwt>(accessToken).organization.slug
  return sdk({ accessToken, organization })
}
