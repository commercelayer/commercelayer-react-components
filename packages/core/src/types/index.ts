export type { RequestConfig } from "./base"
