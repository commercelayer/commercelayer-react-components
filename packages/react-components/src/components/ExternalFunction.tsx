import ExternalFunctionContext, {
  callExternalFunction
} from '#context/ExternalFunctionContext'
import { type ReactNode } from 'react'

interface Props {
  children: ReactNode
  url: string
}

export function ExternalFunction({ children, url }: Props): JSX.Element {
  return (
    <ExternalFunctionContext.Provider value={{ url, callExternalFunction }}>
      {children}
    </ExternalFunctionContext.Provider>
  )
}

export default ExternalFunction
