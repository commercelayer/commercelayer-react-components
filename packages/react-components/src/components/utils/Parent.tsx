import { type Component, type ForwardedRef } from 'react'

export interface ParentProps {
  // Add parent ref to child
  parentRef?: ForwardedRef<any>
  children?: typeof Component | ((P: any) => JSX.Element | null)
}

export default function Parent({
  children,
  ...p
}: ParentProps): JSX.Element | null {
  const Child = children
  return Child !== undefined ? <Child {...p} /> : null
}
