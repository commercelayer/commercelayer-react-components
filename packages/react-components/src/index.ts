export * from '#components/errors/Errors'
export * from '#components/ExternalFunction'
export * from '#components/MetadataInput'
export * from '#components/SubmitButton'
export * from '#components/addresses/Address'
export * from '#components/addresses/AddressCountrySelector'
export * from '#components/addresses/AddressField'
export * from '#components/addresses/AddressInput'
export * from '#components/addresses/AddressStateSelector'
export * from '#components/addresses/AddressesContainer'
export * from '#components/addresses/AddressesEmpty'
export * from '#components/addresses/BillingAddressContainer'
export * from '#components/addresses/BillingAddressForm'
export * from '#components/addresses/SaveAddressesButton'
export * from '#components/addresses/ShippingAddressContainer'
export * from '#components/addresses/ShippingAddressForm'
export * from '#components/auth/CommerceLayer'
export * from '#components/customers/CustomerAddressForm'
export * from '#components/customers/CustomerContainer'
export * from '#components/customers/CustomerField'
export * from '#components/customers/CustomerInput'
export * from '#components/customers/SaveCustomerButton'
export * from '#components/gift_cards/GiftCard'
export * from '#components/gift_cards/GiftCardContainer'
export * from '#components/gift_cards/GiftCardCurrencySelector'
export * from '#components/gift_cards/GiftCardInput'
export * from '#components/gift_cards/GiftCardOrCouponCode'
export * from '#components/gift_cards/GiftCardOrCouponForm'
export * from '#components/gift_cards/GiftCardOrCouponInput'
export * from '#components/gift_cards/GiftCardOrCouponRemoveButton'
export * from '#components/gift_cards/GiftCardOrCouponSubmit'
export * from '#components/line_items/LineItemsContainer'
export * from '#components/line_items/LineItem'
export * from '#components/line_items/LineItemAmount'
export * from '#components/line_items/LineItemCode'
export * from '#components/line_items/LineItemField'
export * from '#components/line_items/LineItemImage'
export * from '#components/line_items/LineItemName'
export * from '#components/line_items/LineItemOption'
export * from '#components/line_items/LineItemOptions'
export * from '#components/line_items/LineItemQuantity'
export * from '#components/line_items/LineItemRemoveLink'
export * from '#components/line_items/LineItemsCount'
export * from '#components/line_items/LineItemsEmpty'
export * from '#components/orders/AddToCartButton'
export * from '#components/orders/AdjustmentAmount'
export * from '#components/orders/CartLink'
export * from '#components/orders/CheckoutLink'
export * from '#components/orders/DiscountAmount'
export * from '#components/orders/GiftCardAmount'
export * from '#components/orders/OrderContainer'
export * from '#components/orders/OrderList'
export * from '#components/orders/OrderListEmpty'
export * from '#components/orders/OrderListRow'
export * from '#components/orders/OrderNumber'
export * from '#components/orders/OrderStorage'
export * from '#components/orders/PaymentMethodAmount'
export * from '#components/orders/PlaceOrderButton'
export * from '#components/orders/PlaceOrderContainer'
export * from '#components/orders/PrivacyAndTermsCheckbox'
export * from '#components/orders/ShippingAmount'
export * from '#components/orders/SubTotalAmount'
export * from '#components/orders/TaxesAmount'
export * from '#components/orders/TotalAmount'
export * from '#components/parcels/ParcelField'
export * from '#components/parcels/ParcelLineItem'
export * from '#components/parcels/ParcelLineItemField'
export * from '#components/parcels/ParcelLineItemsCount'
export * from '#components/parcels/Parcels'
export * from '#components/parcels/ParcelsCount'
export * from '#components/payment_methods/PaymentMethod'
export * from '#components/payment_methods/PaymentMethodName'
export * from '#components/payment_methods/PaymentMethodPrice'
export * from '#components/payment_methods/PaymentMethodRadioButton'
export * from '#components/payment_methods/PaymentMethodsContainer'
export * from '#components/payment_source/PaymentSource'
export * from '#components/payment_source/PaymentSourceBrandIcon'
export * from '#components/payment_source/PaymentSourceBrandName'
export * from '#components/payment_source/PaymentSourceDetail'
export * from '#components/payment_source/PaymentSourceEditButton'
export * from '#components/prices/Price'
export * from '#components/prices/PricesContainer'
export * from '#components/shipments/Shipment'
export * from '#components/shipments/ShipmentField'
export * from '#components/shipments/ShipmentsContainer'
export * from '#components/shipments/ShipmentsCount'
export * from '#components/shipping_methods/ShippingMethod'
export * from '#components/shipping_methods/ShippingMethodName'
export * from '#components/shipping_methods/ShippingMethodPrice'
export * from '#components/shipping_methods/ShippingMethodRadioButton'
export * from '#components/skus/AvailabilityContainer'
export * from '#components/skus/AvailabilityTemplate'
export * from '#components/skus/DeliveryLeadTime'
export * from '#components/skus/SkuField'
export * from '#components/skus/SkuList'
export * from '#components/skus/SkuListsContainer'
export * from '#components/skus/Skus'
export * from '#components/skus/SkusContainer'
export * from '#components/stock_transfers/StockTransfer'
export * from '#components/stock_transfers/StockTransferField'
export * from '#typings/errors'
