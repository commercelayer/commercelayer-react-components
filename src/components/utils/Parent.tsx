import { ForwardedRef, FunctionComponent } from 'react'

export type ParentProps = {
  parentRef?: ForwardedRef<any>
}

const Parent: FunctionComponent<ParentProps> = ({ children, ...p }) => {
  const Child = children as FunctionComponent<ParentProps>
  return Child ? <Child {...p} /> : null
}

export default Parent
