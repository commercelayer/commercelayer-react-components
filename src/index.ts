export { default as AddToCartButton } from '#components/AddToCartButton'
export { default as Address } from '#components/Address'
export { default as AddressCountrySelector } from '#components/AddressCountrySelector'
export { default as AddressField } from '#components/AddressField'
export { default as AddressInput } from '#components/AddressInput'
export { default as AddressStateSelector } from '#components/AddressStateSelector'
export { default as AddressesContainer } from '#components/AddressesContainer'
export { default as AdjustmentAmount } from '#components/AdjustmentAmount'
export { default as AvailabilityContainer } from '#components/AvailabilityContainer'
export { default as AvailabilityTemplate } from '#components/AvailabilityTemplate'
export { default as BillingAddressContainer } from '#components/BillingAddressContainer'
export { default as BillingAddressForm } from '#components/BillingAddressForm'
export { default as CheckoutLink } from '#components/CheckoutLink'
export { default as CommerceLayer } from '#components/CommerceLayer'
export { default as CustomerContainer } from '#components/CustomerContainer'
export { default as CustomerInput } from '#components/CustomerInput'
export { default as DeliveryLeadTime } from '#components/DeliveryLeadTime'
export { default as DiscountAmount } from '#components/DiscountAmount'
export { default as Errors } from '#components/Errors'
export { default as ExternalFunction } from '#components/ExternalFunction'
export { default as GiftCard } from '#components/GiftCard'
export { default as GiftCardAmount } from '#components/GiftCardAmount'
export { default as GiftCardContainer } from '#components/GiftCardContainer'
export { default as GiftCardCurrencySelector } from '#components/GiftCardCurrencySelector'
export { default as GiftCardInput } from '#components/GiftCardInput'
export { default as GiftCardOrCouponCode } from '#components/GiftCardOrCouponCode'
export { default as GiftCardOrCouponForm } from '#components/GiftCardOrCouponForm'
export { default as GiftCardOrCouponInput } from '#components/GiftCardOrCouponInput'
export { default as GiftCardOrCouponRemoveButton } from '#components/GiftCardOrCouponRemoveButton'
export { default as GiftCardOrCouponSubmit } from '#components/GiftCardOrCouponSubmit'
export { default as ItemContainer } from '#components/ItemContainer'
export { default as LineItem } from '#components/LineItem'
export { default as LineItemAmount } from '#components/LineItemAmount'
export { default as LineItemCode } from '#components/LineItemCode'
export { default as LineItemImage } from '#components/LineItemImage'
export { default as LineItemName } from '#components/LineItemName'
export { default as LineItemOption } from '#components/LineItemOption'
export { default as LineItemOptions } from '#components/LineItemOptions'
export { default as LineItemQuantity } from '#components/LineItemQuantity'
export { default as LineItemRemoveLink } from '#components/LineItemRemoveLink'
export { default as LineItemsContainer } from '#components/LineItemsContainer'
export { default as LineItemsCount } from '#components/LineItemsCount'
export { default as LineItemsEmpty } from '#components/LineItemsEmpty'
export { default as MetadataInput } from '#components/MetadataInput'
export { default as OrderContainer } from '#components/OrderContainer'
export { default as OrderNumber } from '#components/OrderNumber'
export { default as OrderStorage } from '#components/OrderStorage'
export { default as PaymentMethod } from '#components/PaymentMethod'
export { default as PaymentMethodAmount } from '#components/PaymentMethodAmount'
export { default as PaymentMethodName } from '#components/PaymentMethodName'
export { default as PaymentMethodPrice } from '#components/PaymentMethodPrice'
export { default as PaymentMethodRadioButton } from '#components/PaymentMethodRadioButton'
export { default as PaymentMethodsContainer } from '#components/PaymentMethodsContainer'
export { default as PaymentSource } from '#components/PaymentSource'
export { default as PaymentSourceBrandIcon } from '#components/PaymentSourceBrandIcon'
export { default as PaymentSourceBrandName } from '#components/PaymentSourceBrandName'
export { default as PaymentSourceDetail } from '#components/PaymentSourceDetail'
export { default as PaymentSourceEditButton } from '#components/PaymentSourceEditButton'
export { default as PlaceOrderButton } from '#components/PlaceOrderButton'
export { default as PlaceOrderContainer } from '#components/PlaceOrderContainer'
export { default as Price } from '#components/Price'
export { default as PricesContainer } from '#components/PricesContainer'
export { default as PrivacyAndTermsCheckbox } from '#components/PrivacyAndTermsCheckbox'
export { default as QuantitySelector } from '#components/QuantitySelector'
export { default as SaveAddressesButton } from '#components/SaveAddressesButton'
export { default as SaveCustomerButton } from '#components/SaveCustomerButton'
export { default as Shipment } from '#components/Shipment'
export { default as ShipmentField } from '#components/ShipmentField'
export { default as ShipmentsContainer } from '#components/ShipmentsContainer'
export { default as ShippingAddressContainer } from '#components/ShippingAddressContainer'
export { default as ShippingAddressForm } from '#components/ShippingAddressForm'
export { default as ShippingAmount } from '#components/ShippingAmount'
export { default as ShippingMethod } from '#components/ShippingMethod'
export { default as ShippingMethodName } from '#components/ShippingMethodName'
export { default as ShippingMethodPrice } from '#components/ShippingMethodPrice'
export { default as ShippingMethodRadioButton } from '#components/ShippingMethodRadioButton'
export { default as Skus } from '#components/Skus'
export { default as SkuContainer } from '#components/SkuContainer'
export { default as SkuField } from '#components/SkuField'
export { default as SkuList } from '#components/SkuList'
export { default as SkuListsContainer } from '#components/SkuListsContainer'
export { default as SkuOption } from '#components/SkuOption'
export { default as SkuOptionInput } from '#components/SkuOptionInput'
export { default as SkuOptionsContainer } from '#components/SkuOptionsContainer'
export { default as StockTransfer } from '#components/StockTransfer'
export { default as StockTransferField } from '#components/StockTransferField'
export { default as SubTotalAmount } from '#components/SubTotalAmount'
export { default as SubmitButton } from '#components/SubmitButton'
export { default as TaxesAmount } from '#components/TaxesAmount'
export { default as TotalAmount } from '#components/TotalAmount'
export { default as VariantSelector } from '#components/VariantSelector'
export { default as VariantsContainer } from '#components/VariantsContainer'
export type { CustomerCardsType } from '#components/utils/PaymentCardsTemplate'
export type { AddressCardsType } from '#components/utils/AddressCardsTemplate'
export type { AddToCartButtonType } from '#components/AddToCartButton'
export type {
  ShippingMethodRadioButtonType,
  ShippingMethodRadioButtonOnChangeType,
} from '#components/ShippingMethodRadioButton'
export type { LineItemCodeType } from '#components/LineItemCode'
export type { LineItemNameType } from '#components/LineItemName'
export type { LineItemImageType } from '#components/LineItemImage'
export type { LineItemType } from '#typings'
export type { ErrorComponentProps, ResourceErrorType } from '#typings/errors'
export type {
  AddressCountrySelectName,
  AddressInputName,
  AddressStateSelectName,
  BaseInputType,
} from '#typings'
