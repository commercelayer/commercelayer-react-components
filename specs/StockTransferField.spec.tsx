import React from 'react'
import { StockTransferField } from '../src'
import renderer from 'react-test-renderer'
import components from '../src/config/components'
import Parent from '../src/components/utils/Parent'

const propTypes = components.StockTransferField.propTypes

test('<StockTransferField/>', () => {
  expect.assertions(4)
  console.error = jest.fn()
  const component = renderer.create(<StockTransferField />)
  const tree = component.toJSON()
  const root = component.toTree()
  const proptypes = root.type['propTypes']
  expect(tree).toMatchSnapshot()
  expect(proptypes.children).toBe(propTypes.children)
  expect(proptypes.type).toBe(propTypes.type)
  expect(console.error.mock.calls[0][2]).toEqual(
    expect.stringContaining(
      'The prop `type` is marked as required in `StockTransferField`, but its value is `undefined`.'
    )
  )
})

test('<StockTransferField check children format />', () => {
  expect.assertions(3)
  const component = renderer.create(<StockTransferField type="quantity" />)
  const tree = component.toJSON()
  const root = component.toTree()
  const rendered = root.rendered
  expect(tree).toMatchSnapshot()
  expect(rendered.type).toBe('p')
  expect(root.props.type).toBe('quantity')
})

test('<StockTransferField with custom children />', () => {
  expect.assertions(6)
  const CustomComponent = (props) => <span>{props.label}</span>
  const component = renderer.create(
    <StockTransferField>{CustomComponent}</StockTransferField>
  )
  const tree = component.toJSON()
  const root = component.toTree()
  const rendered = root.rendered
  const childRendered = root.rendered.rendered
  expect(tree).toMatchSnapshot()
  expect(rendered.type).toBe(Parent)
  expect(childRendered.nodeType).toBe('component')
  expect(childRendered.type).toBe(CustomComponent)
  expect(childRendered.rendered.nodeType).toBe('host')
  expect(childRendered.rendered.type).toBe('span')
})
