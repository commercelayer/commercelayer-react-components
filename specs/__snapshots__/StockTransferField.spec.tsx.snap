// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<StockTransferField check children format /> 1`] = `
<p
  type="quantity"
/>
`;

exports[`<StockTransferField with custom children /> 1`] = `<span />`;

exports[`<StockTransferField/> 1`] = `<p />`;
